concept/answer/ConceptMap.java

```diff
@@ -169,90 +169,110 @@ public int hashCode() {
 
     public static class Explainables {
 
-        Map<Retrievable, Explainable> explainableConcepts;
-        Map<Pair<Retrievable, Retrievable>, Explainable> explainableAttributeOwnerships;
+        private final Map<Retrievable, Explainable> explainableConcepts;
+        private final Map<Pair<Retrievable, Retrievable>, Explainable> explainableOwnerships;
+        private final Map<Retrievable, ? extends Concept> concepts;
 
-        public Explainables() {
-            this(unmodifiableMap(new HashMap<>()), unmodifiableMap(new HashMap<>()));
+        public Explainables(Map<Retrievable, ? extends Concept> concepts) {
+            this(unmodifiableMap(new HashMap<>()), unmodifiableMap(new HashMap<>()), concepts);
         }
 
         public Explainables(Map<Retrievable, Explainable> explainableConcepts,
-                            Map<Pair<Retrievable, Retrievable>, Explainable> explainableAttributeOwnerships) {
-            // to save memory, we only initialise these maps when we have to (we create a lot of concept maps)
+                            Map<Pair<Retrievable, Retrievable>, Explainable> explainableOwnerships,
+                            Map<Retrievable, ? extends Concept> concepts) {
             this.explainableConcepts = explainableConcepts;
-            this.explainableAttributeOwnerships = explainableAttributeOwnerships;
+            this.explainableOwnerships = explainableOwnerships;
+            this.concepts = concepts;
         }
```

Comment:
Make this class static by always constructing the normal concept map first, then call `conceptMap.withExplainables(...)` and `withoutExplanation()`. And do the splitting of `explainableConcepts` in the builder method, before passing into constructor.